lib/umwelt.ex

```diff
@@ -3,16 +3,20 @@ defmodule Umwelt do
   Documentation for `Umwelt`.
   """
 
-  @doc """
-  Hello world.
+  alias Umwelt.Files
+  alias Umwelt.Parser
 
-  ## Examples
-
-      iex> Umwelt.hello()
-      :world
-
-  """
-  def hello do
-    :world
+  def parse_source(project) do
+    Files.list_from_root(project)
+    |> Enum.map(fn
+      filename ->
+        filename
+        |> File.read()
+        |> Parser.read_ast()
+        |> Parser.parse()
+    end)
+    |> Enum.reduce(%{}, fn map, acc ->
+      Map.merge(map, acc)
+    end)
```

Comment:
```suggestion
    project
    |> Files.list_from_root()
    |> Enum.map(& &1 |> File.read() |> Parser.read_ast() |> Parser.parse())
    |> Enum.reduce(&Map.merge/2)
```

Sooner or later it would help if you learned to use an anonymous capture notation, otherwise, the code looks bloated and noisy. Look, `& &` stuff hides the unnecessary part, like parameters and `fn` noise, and reveals what it actually does: read → parse → reduce by merge.

Also: 

- `Enum.reduce/2` fits here, the empty accumulator is redundant,
- `Map.merge/3` would be probably better because `Map.merge/2` silently discards conflicts.

Reply:
I've parsed it! It was hard, but it works )
Looked at `merge/3` but it seems that every module should appear once, if project is correctly designed.

I think about some checks before parsing, at least `mix compile --force --all-warnings` and second defence line is a `Code.string_to_quoted(code)` that ensures syntax is correct.